### PR TITLE
Automatically delete empty groups/communities

### PR DESCRIPTION
**Pull request by Matombo**
Signed-off-by: Werner Sembach <werner.sembach@fau.de>

Another approach to https://github.com/matrix-org/synapse/issues/3023 than https://github.com/matrix-org/synapse/pull/3401.

These changes automatically delete a group/community as soon as the last member leaves the group. The commit also include a sql delta to delete all current empty groups in the database.

### CODE REVIEW

File: synapse/storage/data_stores/main/schema/delta/56/nuke_empty_communities_from_db.sql [hunk 1]
```diff
@@ -0,0 +1,29 @@
+/* Copyright 2019 Matrix.org Foundation CIC
```
Comment (anoadragon453) [inline synapse/storage/data_stores/main/schema/delta/56/nuke_empty_communities_from_db.sql]: This should be copyright yourself :)

File: synapse/groups/groups_server.py [hunk 2]
```diff
@@ -773,6 +773,11 @@ def remove_user_from_group(self, group_id, user_id, requester_user_id, content):
         if not self.hs.is_mine_id(user_id):
             yield self.store.maybe_delete_remote_profile_cache(user_id)
 
+        # Delete group if last user has left
```
Comment (anoadragon453) [inline synapse/groups/groups_server.py]:
```suggestion
        # Delete group if the last user has left
```

**Comment by Matombo** [comment]
Nice, fixed the 2 errors in the comments

**Comment by anoadragon453** [comment]
Wonderful, thank you!

**Pull request merged**